Py_FS/wrapper/nature_inspired/SSA.py

```diff
@@ -0,0 +1,180 @@
+        S = self.algo_params['S'] = float(input('Surface area of body: ') or 0.0154)
+        cd = self.algo_params['cd'] = float(input('Frictional Drag Coefficient: ') or 0.6)
+        CL = self.algo_params['CL'] = float(input('Lift Coefficient: ') or 0.7)
+        hg = self.algo_params['hg'] = float(input('Loss in height occurred after gliding: ') or 1)
+        sf = self.algo_params['sf'] = float(input('Scaling Factor: ') or 18)
+        Gc = self.algo_params['Gc'] = float(input('Gliding Constant: ') or 1.9)
+
+    def initialize(self):
+        super(SSA, self).initialize()
+        self.squirrels_a = initialize_(num_agents=self.num_agents, num_features=self.num_features)
+        self.squirrels_b = initialize_(num_agents=self.num_agents, num_features=self.num_features)
+        self.squirrels_new, self.fitness = sort_agents_(self.squirrels_a, self.fitness)
+        self.ind = np.where(self.fitness == max(self.fitness))[0]
+        D1 = self.algo_params['D1'] = 1 / ((2 * self.algo_params['row'] * self.algo_params['V']) ** (2 * self.algo_params['S'] * self.algo_params['cd']))
+        L = self.algo_params['L'] = 1 / ((2 * self.algo_params['row'] * self.algo_params['V']) ** (2 * self.algo_params['S'] * self.algo_params['CL']))
+        tan_phi = self.algo_params['tan_phi'] = D1 / L
+        dg = self.algo_params['dg'] = self.algo_params['hg'] / (tan_phi * self.algo_params['sf'])
+
+    def check_squirrels(self, s):
+        num_feat = s.shape[1]
+        for i, squirrel in enumerate(s):
+            if np.sum(squirrel) == 0:
+                s[i] = np.random.randint(low=0, high=2, size=(num_feat,))
+        return s
+
+    def get_squirrel(self, agent, alpha=np.random.randint(-2, 3)):
+        features = len(agent)
+        lamb = np.random.uniform(low=-3, high=-1, size=(features))
+        levy = np.zeros((features))
+        get_test_value = 1 / (np.power((np.random.normal(0, 1)), 2))
+        for j in range(features):
+            levy[j] = np.power(get_test_value, lamb[j])  # Eq 2
+        for j in range(features):
+            agent[j] += (alpha * levy[j])  # Eq 1
+        return agent
+
+    def next(self):
+        print('\n================================================================================')
+        print('                          Iteration - {}'.format(self.cur_iter + 1))
+        print('================================================================================\n')
+
+        pdp = self.algo_params['pdp']
+        row = self.algo_params['row']
+        V = self.algo_params['V']
+        S = self.algo_params['S']
+        cd = self.algo_params['cd']
+        CL = self.algo_params['CL']
+        hg = self.algo_params['hg']
+        sf = self.algo_params['sf']
+        Gc = self.algo_params['Gc']
+        D1 = self.algo_params['D1']
+        L = self.algo_params['L']
+        tan_phi = self.algo_params['tan_phi']
+        dg = self.algo_params['dg']

```

Comment:
The next function is called for every iteration. So to reduce redundancy, please move these lines to initialize as recommended previously.

Reply:
@Rohit-Kundu I think moving it isn't necessary. Instead please delete this part and change the parameters in the body accordingly.